src/modules/job-manager/event.c

```diff
@@ -211,6 +219,8 @@ struct event_batch *event_batch_create (struct event *event)
         return NULL;
     if (!(batch->state_trans = json_array ()))
         goto nomem;
+    if (!(batch->annotations = json_array ()))
+        goto nomem;
```

Comment:
One thought would be to leave annotations and state_trans NULL here and create on demand, since a `struct event batch` seems likely to contain only one of the two, unless the system is very busy.